README.md

```diff
@@ -19,3 +19,72 @@ The package can be installed via `pip`:
 ```
 pip install [--user] tuneit
 ```
+
+## Documentation
+
+The Tuneit documentation can be accessed using the link: 
+[tuneit documentation](https://tuneit.readthedocs.io/en/latest/)
+
+The Tuneit package works with computational graphs, which have two main phases:
+
+- A construction phase, where the graph is being built. Every operation that needs to be performed will be added to the graph as node along with all the variables and data input and output. Each type of node is visualised differently in the graph:
+   
+    * **Variables:** they are represented using diamonds. The outline is red in case the variable does not have a value yet and green in  case the variable has been assigned a fixed value.
+    * **Operations:** they are represented using oval shapes.
+    * **Data:** All data objects are represented using rectangles. Most of them represent data inputs, except for the last node in the graph, which represents the data output. 
+
+- A finalization phase. After the graph is finalized, a number of operations can be performed on it:
+
+    * **Visualize:** Using the `visualize()` function the graph can be visualized as it is shown above.
+    * **Compute:** By simply calling the finalized object of the graph, the value final of the graph is computed and returned. 
+    * **Crosshcheck:** The `crosscheck()` function will iterate through all the different options for a variable and return `True` 
+      only for the ones that return the correct result of the graph. 
+    * **Benchmark:** By using the `benchmark()` function, the computation times of all the different combinations of options for the 
+      variables can be compared. In addition, by using the attribute `record` of the function, all those times can be recorded in a  
+      dataframe. Furthermore, the `record` option allows for comparisons between not only the execution times that result by the various 
+      alternatives for the variables, but also different inputs.
+    * **Optimize:** By using the `optimize()` function, the variables of the graph can be tuned. Each time it is called, it returns the values that were used for the variables in that trial and the resulting computation time along with the best trial executed so far. 
+      A trial consists of the timing of an execution of the graph using a different combination of values for the variables that are tuned. 
+
+
+## Example
+
+A small example constructing a graph for the multiplication of a matrix and a vector:
+
+````````````
+@alternatives( 
+    coo = lambda mat: sp.coo_matrix(mat),
+    csc = lambda mat: sp.csc_matrix(mat),
+    csr = lambda mat: sp.csr_matrix(mat),
+    bsr = lambda mat: sp.bsr_matrix(mat)
+)
+def matrix(mat):
+    return s.matrix(mat.todense())
+    
+mat=data(info=["shape","dtype"])
+vec=data(info=["shape","dtype"])
+mat=matrix(mat)
+mul=mat*vec
+mul=finalize(mul)
+
+mul.visualize()
```

Comment:
Here would be nice to have the output of visualize and after of benchmark.
So please split it, add figure here and table after. Then the README it's ok!